cmd/thanos/compact.go

```diff
@@ -786,4 +787,6 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 	cc.webConf.registerFlag(cmd)
 
 	cmd.Flag("bucket-web-label", "External block label to use as group title in the bucket web UI").StringVar(&cc.label)
+
+	cmd.Flag("disable-admin-operations", "Disable UI/API admin operations like marking blocks for deletion and no compaction.").Default("false").BoolVar(&cc.disableAdminOperations)
```

Comment:
can we call it "web.disable-admin-operations" maybe?

Reply:
I was actaully told by @saswatamcode to make it this way. Still I can ofcourse change the flag name, but still how will the PR get merged when these flaky e2e tests don't pass!

Reply:
We could merge without them passing if they are too flaky and make them stable in another PR

Reply:
> can we call it "web.disable-admin-operations" maybe?

btw calling it web.disable-admin-operations doesn't seem to be right as it also disables use of API. and the checks are passing now!! i can get a merge maybe

Reply:
Yup, API & UI both get disabled by the flag, so web prefix is probably not needed. 🙂 

> but still how will the PR get merged when these flaky e2e tests don't pass!

No need to worry so much, there are a few common flakes, which pass with 1/2 retries. I think there are issues to address them.